src/dmd/backend/cod3.d

```diff
@@ -2595,21 +2595,74 @@ code *genmovreg(uint to,uint from)
 
 void genmovreg(ref CodeBuilder cdb,uint to,uint from)
 {
-    debug
-        if (to > ES || from > ES)
-            printf("genmovreg(to = %d, from = %d)\n",to,from);
+    genmovreg(cdb, to, from, TYMAX);
+}
+
+void genmovreg(ref CodeBuilder cdb, uint to, uint from, tym_t tym)
+{
+    // register kind. ex: GPR,XMM,SEG
+    static uint _K(uint reg)
+    {
+        switch (reg)
+        {
+        case ES:                   return ES;
+        case XMM15:
+        case XMM0: .. case XMM7:   return XMM0;
+        case AX:   .. case R15:    return AX;
+        default:                   return reg;
```

Comment:
???? shouldn't this case be a compiler bug?

Reply:
There is an `assert(0)` in line `2661`.